src/screens/DiscoveryScreenV2/fragment/DomainFragment.js

```diff
@@ -127,10 +136,13 @@ const DomainFragment = ({
 
     if (from === FROM_UNFOLLOWED_DOMAIN) {
       const newUnfollowedDomains = [...newMapUnfollowedDomain];
-      newUnfollowedDomains[index].following = !!willFollow;
-      newUnfollowedDomains[index].user_id_follower = myId;
+      const domainIndex = newUnfollowedDomains.findIndex(
+        (domain) => domain.domain_name === item.domain_name
+      );
+      newUnfollowedDomains[domainIndex].following = !!willFollow;
+      // newUnfollowedDomains[domainIndex].user_id_follower = myId;
 
-      const newDomain = newUnfollowedDomains[index];
+      const newDomain = newUnfollowedDomains[domainIndex];
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Again, finding the domain index based on `domain_name` instead of using the array index is a good improvement. This makes the code more robust against changes in the order or number of elements in the `newMapUnfollowedDomain` array.
```diff
- newUnfollowedDomains[index].following = !!willFollow;
- newUnfollowedDomains[index].user_id_follower = myId;
- const newDomain = newUnfollowedDomains[index];
+ const domainIndex = newUnfollowedDomains.findIndex(
+   (domain) => domain.domain_name === item.domain_name
+ );
+ newUnfollowedDomains[domainIndex].following = !!willFollow;
+ // newUnfollowedDomains[domainIndex].user_id_follower = myId;
+ const newDomain = newUnfollowedDomains[domainIndex];
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->